src/Simplify_Let.cpp

```diff
@@ -184,6 +207,9 @@ Body Simplify::simplify_let(const LetOrLetStmt *op, ExprInfo *bounds) {
 
     result = mutate_let_body(result, bounds);
 
+    std::map<std::string, int> vars_used;
+    count_var_uses(result, vars_used);
```

Comment:
This seems like a reasonable algorithm, but don't we now have two independent trackers for var usage? I think info.new_uses and info.old_uses were only used for this purpose. They were inexact, because they tried to run at the same time as the rest of simplification, whereas here you use a post-pass in a separate visitor.

Reply:
(i.e. try deleting new_uses and old_uses entirely if you can)

Reply:
I thought so too, but I saw some issues when I removed the old logic. I'll try again now that you pointed out Loads/Stores can also use a Let.

Reply:
It seems like the info.new_uses/info.old_uses are protecting against some redundant lets getting generated (which trigger the later 'lets must be unique' assert), from *different* `Frame` struct instances.